subjects/localhost/README.md

```diff
@@ -0,0 +1,162 @@
+- You must use the enverement paramatre of cgi. 
+- Your server must receive a request from the browser and send a response using the HTTP header and body
+- Your server must call `select` function (or equivalent function) one time only
+- You must manage chunked and unchunked request.
+- You must set the right status for each response
+
+here is an example of an HTTP request with `GET` method
+
+```http
+GET /Welcome.html HTTP/1.1
+User-Agent: Mozilla/4.0 (compatible; MSIE5.01; Windows NT)
+Host: www.01talent.com
+Accept-Language: en-us
+Accept-Encoding: gzip, deflate
+Connection: Keep-Alive
+```
+and here is an example of an HTTP response : 
+
+```http
+HTTP/1.1 404 Not Found
+Date: Thu, 19 Jul 2022 10:36:20 GMT
+Server: Hserver/1.1.0
+```
+> Stress tests your server. It must stay available at all cost. use this command to test it  `siege -b [IP]:[PORT]`
+	the availability should be up to 95.99
+	
+> Learn about  [CGI](https://en.wikipedia.org/wiki/Common_Gateway_Interface)
+
+
+#### Configurations File
+
+This an example of a simple server configuration.
+```
+server {
+    listen                          80;
+    
+    server_name                     localhost;
+	
+    root                            /usr/share/nginx/html;
+    
+    error_page          Mozilla 404        /usr/share/nginx/errors/404.html;
+    
+    client_body_size                10m; 
+    location /
+    {
+        http_methods  GET  POST DELETE;
+        
+        index index.html
+        
+        upload_status on; //if it's "on," you should add the "upload_path"; otherwise, if it's "off," there is no need for "upload_status." 
+        
+        upload_path         /usr/share/nginx/upload; //depends on "upload_status"
+    }
+}
+```
+
+You have to base your server on this example but You have tested more complex configurations with multiple locations and ports and multiple servers such as :
+```
+{
+server {
+    listen                              80;
+    server_address                      127.0.0.1;
+    server_name                         localhost;
+    ...
+    ...
+    ...
+    location ...{
+
+    }
+    location ...{
+
+    }
+}
+server {
+    listen                              5000;
+    listen                              5001;
+    server_address                      192.182.2.1;
+    server_name                         01edu.com;
+    ...
+    ...
+    ...
+    location ...{
+
+    }
+}
+server {
+    listen                          5001;
+    listen                          5002;
+    listen                          5003;
+    server_address                  192.186.2.54;
+    server_name                     01talent.com;
+    ...
+    ...
+    ...
+    location ...{
+
+    }
+    location ...{
+
+    }
+	location ...{
+
+    }
+}
+}
+```
+The Main Rules:
+- Choose the host(server_address) and a port or multiple ports of each server.
+- The first server for a host: port will be the default if the server_name didn't mutch any other servers server_name.
```

Comment:
- "host:port" without space.
- "match" and not "mutch".
- "any other server `server_name`, no "s" for server and identify server_name as a variable with `\``.